hardware/mifare/mifare_classic.ksy

```diff
@@ -0,0 +1,243 @@
+meta:
+  id: mifare_classic
+  endian: le
+  file-extension: mfd
+  title: "Mifare Classic RFID tag dump"
+  license: BSD-2-Clause
+doc-ref: |
+  https://github.com/nfc-tools/libnfc
+  https://www.nxp.com/docs/en/data-sheet/MF1S70YYX_V1.pdf
+seq:
+  - id: sectors
+    type: sector(_index)
+    repeat: eos
+types:
+  key:
+    seq:
+      - id: key
+        size: 6
+  sector:
+    params:
+      - id: idx
+        type: u1
+    seq:
+      - id: manufacturer
+        type: manufacturer
+        if: has_manufacturer
+      
+      - id: data_
+        -orig-id: abtData
+        size: _io.size - _io.pos - 16 # sizeof(trailer)
+        type: filler
+      - id: trailer
+        type: trailer
+    instances:
+      block_size:
+        value: 16
+      data:
+        value: data_.data
+      blocks:
+        pos: 0
+        io: data_._io
+        size: block_size
+        repeat: eos
+      values:
+        pos: 0
+        io: data_._io
+        type: values
+      has_manufacturer:
+        value: idx == 0
+    types:
+      values:
+        seq:
+          - id: values
+            type: value_block
+            repeat: eos
+        types:
+          value_block:
+            seq:
+              - id: value_
+                type: u4
+                repeat: expr
+                repeat-expr: 3
+              - id: addr_
+                type: u1
+                repeat: expr
+                repeat-expr: 4
+            instances:
+              value_valid:
+                value: 'value_[0]==~value_[1] and value_[0]==value_[2]'
+              addr_valid:
+                value: 'addr_[0]==~addr_[1] and addr_[0]==addr_[2] and addr_[1]==addr_[3]'
+              valid:
+                value: 'value_valid and addr_valid'
+              addr:
+                value: 'addr_[0]'
+                if: valid
+              value:
+                value: value_[0]
+                if: valid
+      filler:
+        doc: "only to create _io"
+        seq:
+          - id: data
+            size: _io.size
```

Comment:
You can just use `filler: {}` or `filler` with only a `doc` element.

Reply:
I found that `data` is needed, but was put into a separate type for getting its `_io`

Reply:
Ok, but then the comment in `doc` is not true anymore.

Reply:
Why not true?

Instead of using filler we could have used its `data` directly with the appropriate size. So it was moved into `filler` only to create `_io`